native/acter/src/api/auth.rs

```diff
@@ -304,23 +305,29 @@ pub async fn register_under_config(
     RUNTIME
         .spawn(async move {
             let client = config.build().await?;
-            if let Err(resp) = client
-                .matrix_auth()
-                .register(register::v3::Request::new())
-                .await
-            {
-                // FIXME: do actually check the registration types...
-                if resp.as_uiaa_response().is_some() {
+            // user registration follows UIAA spec
+            // req-resp exchange continues until final stage of auth process
+            let request = register::v3::Request::new();
+            if let Err(e) = client.matrix_auth().register(request).await {
+                let Some(inf) = e.as_uiaa_response() else {
+                    bail!("No a uiaa response")
+                };
+                if let Some(err) = &inf.auth_error {
+                    bail!("Found auth error: {:?}", err.message);
+                }
+                let has_flow = inf
+                    .flows
+                    .iter()
+                    .any(|f| f.stages.contains(&AuthType::Dummy));
+                let was_completed = inf.completed.contains(&AuthType::Dummy);
+                if has_flow && !was_completed {
```

Comment:
I don't understand what this check is for ... if it doesn't have the flow we are just not continuing? not even showing an error? that seems wrong.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1761/commits/7c3219bcea8563bf8049f8be453e1fc735db229e